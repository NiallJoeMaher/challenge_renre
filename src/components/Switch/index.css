.c-Switch {
  font-family: 'Fredericka the Great', cursive;
  position: relative;
  padding: 16px
}

.c-Switch__label {
  color: #d4d4dc;
  display: inline-block;
  position: relative;
  font-size: 24px;
  top: 50%;
  transform: translateY(-50%); 
}

.c-Switch__label--active {
  color: #feda6a;
  display: inline-block;
  position: relative;
  font-size: 24px;
  top: 50%;
  transform: translateY(-50%); 
}


.c-Switch__toggle-btn {
  box-sizing: initial;
  display: inline-block;
  outline: 0;
  width: 60px;
  height: 30px;
  position: relative;
  cursor: pointer;
  user-select: none;
  background: #feda6a;;
  border-radius: 4em;
  padding: 4px;
  transition: all 0.4s ease;
  border: 2px solid #d4d4dc;
  margin: 0 16px;
}
.c-Switch__toggle-input:focus + .c-Switch__toggle-btn::after,
.c-Switch__toggle-btn:active::after {
  box-sizing: initial;
  box-shadow: 0 0 0 2px rgba(0, 0, 0, 0.1),
    0 4px 0 rgba(0, 0, 0, 0.08), inset 0px 0px 0px 3px #9c9c9c;
}
.c-Switch__toggle-btn::after {
  left: 0;
  position: relative;
  display: block;
  content: '';
  width: 50%;
  height: 100%;
  border-radius: 4em;
  background: #d4d4dc;
  transition: all 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275),
    padding 0.3s ease, margin 0.3s ease;
  box-shadow: 0 0 0 2px rgba(0, 0, 0, 0.1),
    0 4px 0 rgba(0, 0, 0, 0.08);
}
.c-Switch__toggle-btn.c-Switch--toggle-btn-on::after {
  left: 50%;
}

.c-Switch__toggle-btn.c-Switch--toggle-btn-on:active {
  box-shadow: none;
}
.c-Switch__toggle-btn.c-Switch--toggle-btn-on:active::after {
  margin-left: -1.6em;
}
.c-Switch__toggle-btn:active::after {
  padding-right: 1.6em;
}
.c-Switch__toggle-input {
  /* visually hidden but still accessible */
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  white-space: nowrap;
}