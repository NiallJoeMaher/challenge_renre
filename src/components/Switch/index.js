import React, { Component } from 'react';
import './index.css';

import PropTypes from 'prop-types';

const noop = () => {};

class Switch extends Component {
  static propTypes = {
    leftLabel: PropTypes.string,
    rightLabel: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    on: PropTypes.bool.isRequired
  };

  render() {
    const className = 'c-Switch';
    const {
      on,
      'aria-label': ariaLabel,
      onClick,
      leftLabel,
      rightLabel
    } = this.props;
    const btnClassName = [
      `${className}__toggle-btn`,
      on ? `${className}--toggle-btn-on` : `${className}--toggle-btn-off`
    ]
      .filter(Boolean)
      .join(' ');

    return (
      <div className={className}>
        <label aria-label={ariaLabel || 'Toggle'}>
          <input
            className={`${className}__toggle-input`}
            type='checkbox'
            checked={on}
            onChange={noop}
            onClick={onClick}
            data-testid='toggle-input'
          />
          <span className={`${className}__label${!on ? '--active' : ''}`}>
            {leftLabel}
          </span>
          <span className={btnClassName} />
          <span className={`${className}__label${on ? '--active' : ''}`}>
            {rightLabel}
          </span>
        </label>
      </div>
    );
  }
}

export default Switch;
