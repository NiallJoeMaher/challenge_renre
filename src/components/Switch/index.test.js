import React from 'react';
import { shallow } from 'enzyme';

import Switch from './';

it('renders as expected', () => {
  const component = shallow(
    <Switch
      leftLabel='&deg;C'
      rightLabel='&deg;F'
      onClick={() => {}}
      on={false}
    />
  );
  expect(component).toMatchSnapshot();
});

it('renders as on as expected', () => {
  const component = shallow(
    <Switch
      leftLabel='&deg;C'
      rightLabel='&deg;F'
      onClick={() => {}}
      on={true}
    />
  );
  expect(component).toMatchSnapshot();
});

it('Test click event', () => {
  const mockCallBack = jest.fn();
  const button = shallow(
  <Switch
    leftLabel='&deg;C'
    rightLabel='&deg;F'
    onClick={mockCallBack}
    on={true}
  />);
  button.find('.c-Switch__toggle-input').simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
