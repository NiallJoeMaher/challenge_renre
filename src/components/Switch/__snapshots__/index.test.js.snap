// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="c-Switch"
>
  <label
    aria-label="Toggle"
  >
    <input
      checked={false}
      className="c-Switch__toggle-input"
      data-testid="toggle-input"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="c-Switch__label--active"
    >
      째C
    </span>
    <span
      className="c-Switch__toggle-btn c-Switch--toggle-btn-off"
    />
    <span
      className="c-Switch__label"
    >
      째F
    </span>
  </label>
</div>
`;

exports[`renders as on as expected 1`] = `
<div
  className="c-Switch"
>
  <label
    aria-label="Toggle"
  >
    <input
      checked={true}
      className="c-Switch__toggle-input"
      data-testid="toggle-input"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="c-Switch__label"
    >
      째C
    </span>
    <span
      className="c-Switch__toggle-btn c-Switch--toggle-btn-on"
    />
    <span
      className="c-Switch__label--active"
    >
      째F
    </span>
  </label>
</div>
`;
